services/management/external.go

```diff
@@ -82,6 +82,10 @@ func (e ExternalService) AddExternal(ctx context.Context, req *managementpb.AddE
 		}
 		res.Service = invService.(*inventorypb.ExternalService)
 
+		if isPushMode(req.MetricsMode) {
+			req.MetricsMode = managementpb.MetricsMode_PULL
```

Comment:
Independently from the comment above, *we must not overwrite mode if it is not auto*. If requested mode is not auto and not supported, we must return error. Everywhere.